docs/standard/io/buffers.md

```diff
@@ -0,0 +1,188 @@
+- Two `SequencePosition` can't be compared, making it difficult to:
+  - Know if one position is greater than or less than another position.
+  - Write some parsing algorithms.
+- `ReadOnlySequence<T>` is bigger than an object reference and should be passed by [in](/dotnet/csharp/language-reference/keywords/in-parameter-modifier) or [ref](/dotnet/csharp/language-reference/keywords/ref) where possible. Passing `ReadOnlySequence<T>` by `in` or `ref` reduces copies of the [struct](../../csharp/language-reference/keywords/struct.md).
+- Empty segments:
+  - Are valid within a `ReadOnlySequence<T>`
+  - Can appear when iterating using `ReadOnlySequence<T>.TryGet` 
+  - Can appear slicing the sequence using `ReadOnlySequence<T>.Slice()` with `SequencePosition`(s).
+
+## SequenceReader\<T\>
+
+<xref:System.Buffers.SequenceReader%601>:
+
+- Is a new type that was introduced in .NET Core 3.0 to simplify the processing of a `ReadOnlySequence<T>`.
+- Unifies the differences between a single segment `ReadOnlySequence<T>` and multi-segment `ReadOnlySequence<T>`.
+- Provides helpers for reading binary and text data (`byte` and `char`) that may or may not be split across segments.
+
+There are built-in methods for dealing with processing both binary and delimited data. The following section demonstrates what those same methods look like with the `SequenceReader<T>`:
+
+### Access data
+
+`SequenceReader<T>` has methods for enumerating data inside of the `ReadOnlySequence<T>` directly. The following code is an example of processing a `ReadOnlySequence<byte>` a `byte` at a time:
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet8)]
+
+The `CurrentSpan` exposes the current segment's `Span`, which is similar to what was done in the method manually.
+
+### Position
+
+The following code is an example implementation of `FindIndexOf` using the `SequenceReader<T>`:
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet9)]
+
+### Process binary data
+
+The following example parses a 4-byte big-endian integer length from the start of the `ReadOnlySequence<byte>`.
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet11)]
+
+### Process text data
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet10)]
+
+### SequenceReader\<T\> problem areas
+
+- `SequenceReader<T>` is a mutable struct, it should always be passed by [reference](../../csharp/language-reference/keywords/ref.md).
+- `SequenceReader<T>` is a [ref struct](../../csharp/language-reference/keywords/ref.md#ref-struct-types) so it can only be used in synchronous methods and can't be stored in fields. For more information, see [Write safe and efficient C# code](../../csharp/write-safe-efficient-code.md).
+- `SequenceReader<T>` is a forward-only reader, and `Advance` doesn't support negative numbers.
```

Comment:
It is intended to _primarily_ be a forward reader. You can `Rewind()`, but code will potentially be more efficient if you don't call it as it may involve seeking from the start of the `Sequence` (if you rewind past the current span).

Reply:
@JeremyKuhne suggest an edit plz

Reply:
`SequenceReader<T>` is optimized for use as a forward-only reader. `Rewind()` is intended for small backups that can't be addressed utilizing other `Read`, `Peek`, and `IsNext` APIs.